### PR TITLE
Add pagination for teams page

### PR DESCRIPTION
**Pull request by mayankagnihotri7**
## Notion Card
[Notion Card](https://www.notion.so/saeloun/Implement-pagination-on-teams-page-2d6167ca653a42b3bc240c415f05f89d)

## Description
- Added pagination for the teams' page.

## Preview

https://github.com/saeloun/miru-web/assets/57438322/e6282625-f059-4733-9367-3d769fd5262a

### CODE REVIEW

File: app/controllers/internal_api/v1/team_controller.rb [hunk 1]
```diff
@@ -11,7 +11,16 @@ def index
       .ransack(first_name_or_last_name_or_recipient_email_cont: params.dig(:q, :first_name_or_last_name_or_email_cont))
     teams = query.result(distinct: true)
     invitations = invitations_query.result(distinct: true)
-    render :index, locals: TeamPresenter.new(teams, invitations, current_user, current_company).index_data, status: :ok
+    presenter_data = TeamPresenter.new(teams, invitations, current_user, current_company).index_data
```
Comment (prasanthchaduvula) [inline app/controllers/internal_api/v1/team_controller.rb:14]: Add an empty line before between `invitations` and `presenter_data`

File: app/controllers/internal_api/v1/team_controller.rb [hunk 2]
```diff
@@ -11,7 +11,16 @@ def index
...
+    team_data = presenter_data[:teams]
+    invitation_data = presenter_data[:invitations]
+    pagy_details, team_details = pagy_array(team_data, items: params[:items] || 10)
+    invitation_pagy_data, invitation_details = pagy_array(invitation_data, items: params[:items] || 5)
+    render :index, locals: {
```
Comment (prasanthchaduvula) [inline app/controllers/internal_api/v1/team_controller.rb:19]: Add an empty line before `render`

File: app/javascript/src/common/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -121,10 +123,18 @@ const Pagination = ({
           <select
             className="p-2 text-xs font-bold text-miru-han-purple-1000"
             defaultValue={pagy?.items}
-            value={params.invoices_per_page}
-            onChange={e =>
-              setParams({ page: 1, invoices_per_page: Number(e.target.value) })
-            }
+            value={isTeamPage ? params?.items : params.invoices_per_page}
+            onChange={e => {
+              if (isTeamPage) {
```
Comment (prasanthchaduvula) [inline app/javascript/src/common/Pagination/Pagination.tsx:128]: move this logic to a function

**Review by prasanthchaduvula — CHANGES_REQUESTED** [review]

**Comment by mayankagnihotri7** [comment]
@prasanthchaduvula I've made the requested changes. Please review.

**Review by prasanthchaduvula — APPROVED** [review]

**Comment by apoorv1316** [comment]
@mayankagnihotri7 Please watch loom video and fix the bugs mentioned in video https://www.loom.com/share/63d6d3c3ae2a472b823795b4843bd721?focus_title=1&muted=1&from_recorder=1

**Comment by mayankagnihotri7** [comment]
@prasanthchaduvula Can you please review the code again? I've made some changes.

@apoorv1316 I've made the requested changes. Please review.

**Review by apoorv1316 — APPROVED** [review]

**Comment by apoorv1316** [comment]
@mayankagnihotri7 Please add system specs for it.

**Pull request merged**